tribits/core/package_arch/TribitsGitRepoVersionInfo.cmake

```diff
@@ -44,6 +44,74 @@
 ################################################################################
 
 
+# @FUNCTION: tribits_git_repo_sha1()
+#
+# Get the Git repo version SHA1.
+#
+# Usage::
+#
+#   tribits_git_repo_sha1(<gitRepoDir> <gitRepoSha1Out>
+#     FAILURE_MESSAGE_OUT <failureMsgOut>)
+#
+# If ``<failureMsgOut>`` is non-empty on input, then the variable
+# ``<failureMsgOut>`` will be set to a non-empty value on output with the
+# error message if an error occurs and the SHA1 for ``<gitRepoDir>`` can not
+# be computed.  If ``<failureMsgOut>`` is empty in input, and there is an
+# error, a ``FATAL_ERROR`` will be raised.
+#
+# NOTE: To get the SHA1 for the Git repo ``<gitRepoDir>``, it must contain the
+# ``.git/`` directory and the ``git`` executable var ``GIT_EXECUTABLE`` must
+# be set.  Otherwise, it is an error.
+#
+function(tribits_git_repo_sha1  gitRepoDir  gitRepoSha1Out)
+
+  cmake_parse_arguments( PARSE_ARGV 2
+    PARSE "" "" # prefix, options, one_value_keywords
+    "FAILURE_MESSAGE_OUT" # multi_value_keywords
+    )
+  tribits_check_for_unparsed_arguments()
+  tribits_assert_parse_arg_zero_or_one_value(PARSE  FAILURE_MESSAGE_OUT)
+
+  set(failureMsg "")
+
+  if (NOT GIT_EXECUTABLE)
```

Comment:
Can you use `Git::Git` in the `COMMAND` for `execute_process()`?  There is no mention that you can in the documentation:

* https://cmake.org/cmake/help/v3.23/command/execute_process.html

Reply:
The advantage of using the variable `GIT_EXECUTABLE` instead of the target `Git::Git`` is that I can test that function in a `ctest -P` unit test script.  You can't define targets in a ctest -P script.

Reply:
OK, so this is in a `-P` script. In that case using `GIT_EXECUTABLE` would be best. You may still want to check out `FindGit` though.

Reply:
This module can be used in a cmake -P script and the unit test driver that runs some unit tests on this fucntion is run with cmake -P.

What does `find_package()` do when called from a cmake -P script where targets are not allowed?

Reply:
It is legal to call `find_package()` from inside `cmake -P` as long as the package doesn't try to create a target. [`FindGit`](https://cmake.org/cmake/help/latest/module/FindGit.html) specifically has a provision to not attempt to create the `Git::Git` target if it's called from inside `cmake -P`:

> *New in version 3.14:* The module defines the following `IMPORTED` targets (when [`CMAKE_ROLE`](https://cmake.org/cmake/help/latest/prop_gbl/CMAKE_ROLE.html#prop_gbl:CMAKE_ROLE) is `PROJECT`):
>
> `Git::Git`
>
> Executable of the Git command-line client.

Reply:
We will keep using `GIT_EXECUTABLE` instead of `Git::Git` since this is called in cmake -P scripts as well.